common/test/acceptance/tests/lms/test_progress_page.py

```diff
@@ -0,0 +1,304 @@
+
+    @ddt.data(
+        _edit_problem_content,
+        _change_subsection_structure,
+        _change_weight_for_problem,
+        _rescore_for_all
+    )
+    def test_content_changes_do_not_change_score(self, edit):
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(0, 1)])
+            self.assertEqual(self._get_section_score(), (0, 1))
+            self.courseware_page.visit()
+            self._answer_problem_correctly()
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+        edit(self)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+    def test_visibility_change_does_affect_score(self):
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(0, 1)])
+            self.assertEqual(self._get_section_score(), (0, 1))
+            self.courseware_page.visit()
+            self._answer_problem_correctly()
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+        self._set_staff_lock_on_subsection(True)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), None)
+            self.assertEqual(self._get_section_score(), None)
+
+        self._set_staff_lock_on_subsection(False)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+
+class ExplicitGradedFieldTest(ProgressPageBaseTest):
+    """
+    Tests changing a subsection's 'graded' field
+    and the effect it has on the progress page.
+    """
+    def setUp(self):
+        super(ExplicitGradedFieldTest, self).setUp()
+        self._set_policy_for_subsection("Homework")
+
+    def test_explicit_graded_field_does_not_affect_score(self):
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(0, 1)])
+            self.assertEqual(self._get_section_score(), (0, 1))
+            self.assertTrue(self.progress_page.text_on_page("Homework 1 - Test Subsection 1 - 0% (0/1)"))
+            self.courseware_page.visit()
+            self._answer_problem_correctly()
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+            self.assertTrue(self.progress_page.text_on_page("Homework 1 - Test Subsection 1 - 100% (1/1)"))
+
+        self._set_policy_for_subsection("Not Graded")
+
+        with self._logged_in_session():
+            self.progress_page.visit()
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+            self.assertFalse(self.progress_page.text_on_page("Homework 1 - Test Subsection 1"))
```

Comment:
It's actually the path going from "Not Graded" to "Graded" that wasn't working.  So we should add that in addition to, or instead of.